### PR TITLE
fix(ffmpeg): add native filesystem API support for FFmpeg image decoder

### PR DESCRIPTION
**Pull request by lxydiy**
#7191 , but use AVIO for better compatibility.  

@FASTSHIFT 

**Edit:**
Tested with .jpg .png on linux 5.10 
| library | version |
|---|---|
| libavutil | 56.70.100 |
| libavcodec | 58.134.100 |
| libavformat | 58.76.100 |
| libswscale | 5.9.100 |

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by diylxy** [comment]
You can pass `true` to [this](https://github.com/lvgl/lvgl/pull/7253/files#diff-d07edb0d56b3b2bf058501d27bc0c4bb76edf83ed9bffa3307dae6bfd4543902R138) as well as [this](https://github.com/lvgl/lvgl/pull/7253/files#diff-d07edb0d56b3b2bf058501d27bc0c4bb76edf83ed9bffa3307dae6bfd4543902R169) parameters to use native path like `A:/video.mp4` in ffmpeg_player widget. I set this parameter `false` for some kinds of [backward capability](https://github.com/lvgl/lvgl/blob/f94b34fff12a4c6442f11a394f582296b1a6a140/docs/details/libs/ffmpeg.rst?plain=1#L36-L38).

File: src/libs/ffmpeg/lv_ffmpeg.c [hunk 1]
```diff
@@ -291,7 +295,7 @@ static lv_result_t decoder_open(lv_image_decoder_t * decoder, lv_image_decoder_d
     if(dsc->src_type == LV_IMAGE_SRC_FILE) {
         const char * path = dsc->src;
 
-        struct ffmpeg_context_s * ffmpeg_ctx = ffmpeg_open_file(path);
+        struct ffmpeg_context_s * ffmpeg_ctx = ffmpeg_open_file(path, &dsc->file);
```
Comment (FASTSHIFT) [inline src/libs/ffmpeg/lv_ffmpeg.c]: `ffmpeg_open_file(path, true);` ?

**Comment by FASTSHIFT** [comment]
> You can pass `true` to [this](https://github.com/lvgl/lvgl/pull/7253/files#diff-d07edb0d56b3b2bf058501d27bc0c4bb76edf83ed9bffa3307dae6bfd4543902R138) as well as [this](https://github.com/lvgl/lvgl/pull/7253/files#diff-d07edb0d56b3b2bf058501d27bc0c4bb76edf83ed9bffa3307dae6bfd4543902R169) parameters to use native path like `A:/video.mp4` in ffmpeg_player widget. I set this parameter `false` for some kind of [backward capability](https://github.com/lvgl/lvgl/blob/f94b34fff12a4c6442f11a394f582296b1a6a140/docs/details/libs/ffmpeg.rst?plain=1#L36-L38).

You can use macro switches to control forward compatibility (the default value uses lv_fs mode). You can recommend users to use a unified lv_fs file path mode in the documentation to avoid confusion in file path management.

File: src/libs/ffmpeg/lv_ffmpeg.c (hunk 1 above)
Comment (lxydiy) [reply src/libs/ffmpeg/lv_ffmpeg.c]: fixed.
What surprised me was that no compiler detected this error.

**Comment by diylxy** [comment]
> You can use macro switches to control forward compatibility (the default value uses lv_fs mode). You can recommend users to use a unified lv_fs file path mode in the documentation to avoid confusion in file path management.

Added an option, but doing so will affect the playback of online video streams, so it's disabled by default. 
BTW, the document may need further improvement as I am not familiar with dealing with `.rst` documents.

**Review by FASTSHIFT — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**